flink-table-store-core/src/main/java/org/apache/flink/table/store/file/operation/FileStoreWrite.java

```diff
@@ -52,4 +56,35 @@ RecordWriter<T> createEmptyWriter(
      */
     Callable<CompactResult> createCompactWriter(
             BinaryRowData partition, int bucket, @Nullable List<DataFileMeta> compactFiles);
+
+    /**
+     * If overwrite is true, the writer will overwrite the store, otherwise it won't.
+     *
+     * @param overwrite the overwrite flag
+     */
+    void withOverwrite(boolean overwrite);
+
+    /**
+     * Write the record to the store.
+     *
+     * @param record the given record
+     * @throws Exception the thrown exception when writing the record
+     */
+    void write(SinkRecord record) throws Exception;
```

Comment:
`SinkRecord` is a class in `table` instead of file store.
Can we just introduce a  `write(BinaryRowData partition, int bucket, T t)` to this?
`WriteFunction` can be hold in `TableWriteImpl` instead of `AbstractFileStoreWrite`.

Reply:
Thanks @JingsongLi and it looks good to me, I have updated the codes, thx